paimon-web-server/src/main/java/org/apache/paimon/web/server/service/impl/SessionServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+import org.apache.paimon.web.server.service.SessionService;
+import org.apache.paimon.web.server.service.UserService;
+import org.apache.paimon.web.server.service.UserSessionManager;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+/** The implementation of {@link SessionService}. */
+@Service
+public class SessionServiceImpl implements SessionService {
+
+    private static final Integer ACTIVE_STATUS = 1;
+    private static final Integer INACTIVE_STATUS = 0;
+
+    @Autowired private UserSessionManager sessionManager;
+
+    @Autowired private UserService userService;
+
+    @Override
+    public void createSession(SessionDTO sessionDTO) {
+        try {
+            SqlGatewayClient client =
+                    new SqlGatewayClient(sessionDTO.getHost(), sessionDTO.getPort());
+            if (sessionDTO.getUid() != null) {
+                String sessionName;
+                String username = userService.getUserById(sessionDTO.getUid()).getUsername();
+                if (StringUtils.isNotBlank(sessionDTO.getName())) {
+                    sessionName = sessionDTO.getName();
+                } else {
+                    sessionName = username + "_" + UUID.randomUUID();
+                }
+                SessionEntity sessionEntity = client.openSession(sessionName);
+                sessionManager.addSession(
+                        sessionDTO.getUid() + "_" + sessionDTO.getClusterId(), sessionEntity);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create session", e);
+        }
+    }
+
+    @Override
+    public void closeSession(SessionDTO sessionDTO) {
+        try {
+            SqlGatewayClient client =
+                    new SqlGatewayClient(sessionDTO.getHost(), sessionDTO.getPort());
+            if (sessionDTO.getUid() != null) {
+                SessionEntity session =
+                        sessionManager.getSession(
+                                sessionDTO.getUid() + "_" + sessionDTO.getClusterId());
+                if (session != null) {
+                    client.closeSession(session.getSessionId());
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to close session", e);
+        }
+    }
+
+    @Override
+    public int triggerSessionHeartbeat(SessionDTO sessionDTO) {
+        try {
+            if (sessionDTO.getUid() != null) {
+                SqlGatewayClient client =
+                        new SqlGatewayClient(sessionDTO.getHost(), sessionDTO.getPort());
+                SessionEntity session =
+                        sessionManager.getSession(
+                                sessionDTO.getUid() + "_" + sessionDTO.getClusterId());
+                client.triggerSessionHeartbeat(session.getSessionId());
```

Comment:
Where does the session id store?

Reply:
Stored in a map.
 private final ConcurrentHashMap<String, SessionEntity> sessions = new ConcurrentHashMap<>();

Reply:
@s7monk, when the server is shutdown due to exception, what's the behavior of the heartbeat?

Reply:
If the server shuts down abnormally, the current implementation should continue to send heartbeats. Is it necessary to add counting logic to stop sending heartbeats after the number of heartbeat failures?

Reply:
> @s7monk, when the server is shutdown due to exception, what's the behavior of the heartbeat?

If the server shuts down abnormally, the current implementation should continue to send heartbeats. Is it necessary to add counting logic to stop sending heartbeats after the number of heartbeat failures?